lib/p2p/ghost_protocol/message.ex

```diff
@@ -148,15 +148,17 @@ defmodule Elixium.P2P.GhostProtocol.Message do
   # by the length, in bytes, of the rest of the message
   @spec parse_header(reference) :: {String.t(), integer}
   defp parse_header(socket) do
-    {:ok, header} =
-      socket
-      # Will get "Ghost|00000000|v1.0|" from socket
-      |> :gen_tcp.recv(20)
-
-    [protocol, bytes, version, predata] = String.split(header, "|")
-    {bytes, _} = Integer.parse(bytes)
-
-    {protocol, bytes, predata}
+    case :gen_tcp.recv(socket, 20) do
```

Comment:
Fixes an error where we'd lose connection and not know it, then the handler would crash. This just makes the handler exit in a normal and safe way